<launch>
    <arg name="nao_ip" default="195.168.1.4" /> <!-- IP address of non-simulated robot (ignored if simulated) -->
    <arg name="use_sim_nao" default="true" /> <!-- if true, webots simulation expected to be running -->

    <arg name="nao_handedness" default="right" />
    <arg name="writing_surface_frame_id" default="writing_surface" /> 
    <arg name="trajectory_topic" default="write_traj" /> 
    <arg name="clear_writing_surface_topic" default="clear_screen" /> 
    <arg name="shape_writing_finished_topic" default="shape_finished" />
    <arg name="user_drawn_shapes_topic" default="user_drawn_shapes" />

    <arg name="nao_ip_to_use" value="127.0.0.1" if="$(arg use_sim_nao)" /> <!-- look for local simulator -->
    <arg name="nao_ip_to_use" value="$(arg nao_ip)" unless="$(arg use_sim_nao)" />

    <!-- Start the message echoer which will listen for a message and request that the Nao traces it -->
    <node pkg="nao_trajectory_following" type="message_echoer_nao.py" name="message_echoer">   
        <param name="nao_handedness" value="$(arg nao_handedness)" />
        <param name="nao_ip" value="$(arg nao_ip_to_use)"/>
        <param name="writing_surface_frame_id" value="$(arg writing_surface_frame_id)" /> 
        
        <param name="shape_writing_finished_topic" type="str" value="$(arg shape_writing_finished_topic)"/>
        <param name="user_drawn_shapes_topic" type="str" value="$(arg user_drawn_shapes_topic)"/>
        <param name="trajectory_output_topic" type="str" value="$(arg trajectory_topic)"/>
        <param name="clear_writing_surface_topic" value="$(arg clear_writing_surface_topic)" />
    </node>
    
    <!-- Start the capabilities for the Nao to trace messages on a writing surface -->
    <include file="$(find nao_trajectory_following)/launch/nao_writing_on_surface.launch" >
        <arg name="use_sim_nao" value="$(arg use_sim_nao)" />
        <arg name="nao_handedness" value="$(arg nao_handedness)" />
        <arg name="nao_ip" value="$(arg nao_ip_to_use)"/>
        <arg name="writing_surface_frame_id" value="$(arg writing_surface_frame_id)" /> 
        <arg name="trajectory_visualization_input_topic" value="$(arg trajectory_topic)" />
        <arg name="trajectory_nao_input_topic" value="$(arg trajectory_topic)" />
        <arg name="clear_writing_surface_topic" value="$(arg clear_writing_surface_topic)" />
    </include>
</launch>
